mitodl/ocw-hugo-themes
# add a script for building an entire output folder from ocw-to-hugo

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/5

#### What's this PR do?
On the OCW Next build servers currently, we use the `build.sh` script in `ocw-course-hugo-starter` to iterate and build all the courses output by `ocw-to-hugo`.  Since we are moving to using `ocw-hugo-themes`, we will need a replacement for this script.  This PR adds `build_all_courses.sh` into the `build_scripts` folder to serve this purpose

#### How should this be manually tested?
 - Clone `ocw-to-hugo` and follow the readme to convert a number of courses more than 1, the example courses should be fine
 - Clone `ocw-www`  and check out the `cg/content-only` branch
 - Set `EXTERNAL_SITE_PATH` to the path to `ocw-www`
 - Run `npm start`
 - Set the new env variables for `build_all_courses.sh`:
   - `OCW_TO_HUGO_OUTPUT_DIR`: The output directory you configured in `ocw-to-hugo`
   - `COURSE_OUTPUT_DIR`: The path you want to output the courses to, in this case your `/path/to/ocw-www/public/courses`
   - `COURSE_BASE_URL`: The `--baseUrl` argument to send Hugo for each course build, in this case `http://localhost:3000/courses`
   - `VERBOSE`: `1` if you want to see output from the builds, otherwise `0`
 - In a new terminal, run `./build_scripts/build_all_courses.sh`
 - Once the builds are complete, try and visit a course site for one of the courses you built, at a URL like http://localhost:3000/courses/18-06-linear-algebra-spring-2010
